tests/test_click_odoo.py

```diff
@@ -235,6 +235,55 @@ def testcmd(env):
     assert "No database provided" in result.output
 
 
+def test_env_options_addons_path(odoodb, tmpdir):
+    addons_path = ','.join([
+        os.path.join(odoo.__path__[0], 'addons'),
+        os.path.join(odoo.__path__[0], '..', 'addons'),
+        os.path.join(os.path.dirname(__file__), 'data', 'addons'),
+    ])
+
+    @click.command()
+    @click_odoo.env_options()
+    def testcmd(env):
+        login = env['res.users'].search([('login', '=', 'admin')]).login
+        click.echo("login={}".format(login))
+
+    # addons path from command line
+    runner = CliRunner()
+    result = runner.invoke(testcmd, [
+        '-d', odoodb,
+        '--addons-path', addons_path,
+    ])
+    assert result.exit_code == 0
+    assert 'login=admin\n' in result.output
+    # addons path in config
+    odoocfg1 = tmpdir / 'odoo1.cfg'
+    odoocfg1.write(textwrap.dedent("""\
+        [options]
+        db_name={}
+        addons_path={}
+    """.format(odoodb, addons_path)))
+    result = runner.invoke(testcmd, [
+        '-c', str(odoocfg1,),
+    ])
+    assert result.exit_code == 0
+    assert 'login=admin\n' in result.output
+    # --addons-path has priority over addons_path in config
+    odoocfg2 = tmpdir / 'odoo2.cfg'
+    odoocfg2.write(textwrap.dedent("""\
+        [options]
+        db_name=notadb
+        addons_path=notanaddonspath
+    """))
+    result = runner.invoke(testcmd, [
+        '-c', str(odoocfg2),
+        '-d', odoodb,
+        '--addons-path', addons_path,
+    ])
+    assert result.exit_code == 0
+    assert 'login=admin\n' in result.output
```

Comment:
This test does not seem to test the addons path is actually taken into account.
There is no need to do a `search` in this test, nor do we need the `odoodb` fixture.

I suggest writing a `script5.py` which simply does `import odoo.addons.addon1` and call it using subprocess (`click-odoo -- script5.py`) in the test to check if it succeeds or not depending on the addons path (see other subprocess tests).

Reply:
Much better :+1: